File: app/src/main/java/com/example/imhere/MainActivity.java
package com.example.imhere;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import android.Manifest;
import android.annotation.SuppressLint;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.location.Location;
import android.location.LocationListener;
import android.location.LocationManager;
import android.net.wifi.WifiInfo;
import android.net.wifi.WifiManager;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.TabHost;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.TimeUnit;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    //region field
    String[] name = {"Как дела", "Универ", "Преподы", "Кто ты", "За путина", "Опрос", "Радик", "Радик"};
    String[] type = {"временный", "постоянный", "временный", "постоянный", "временный", "постоянный", "временный", "временный"};
    String[] who = {"радик", "Универ", "Преподы", "союз", "деканат", "декан", "преподы", "преподы"};
    String[] time = {"12:00", "не ограничено", "12:00", "не ограничено", "12:00", "не ограничено", "12:00", "12:00"};

    final String ATTRIBUTE_NAME_NAME = "textName";
    final String ATTRIBUTE_NAME_TYPE = "textType";
    final String ATTRIBUTE_NAME_WHO = "textWho";
    final String ATTRIBUTE_NAME_TIME = "textTime";
    final String ATTRIBUTE_NAME_IMG = "image";

    ListView listViewInterview;

    public Button checkButton;
    public Button exitButton;

    public TextView locationText;
    public TextView wifiText;

    public ImageView classImage;
    public TextView classNameText;
    public TextView classNumberText;
    public TextView classTypeText;
    public TextView auditoryText;
    public TextView lecturerText;
    public TextView timeText;

    private LocationManager locationManager;
    public Location locationStudent;
    private Location locationRTF;

    WifiInfo wifiInfo;
    //endregion

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //region Initializing
        checkButton = findViewById(R.id.checkButton);
        exitButton = findViewById(R.id.exitButton);

        checkButton.setOnClickListener(this);
        exitButton.setOnClickListener(this);
        //endregion

        //region Location
        locationText = findViewById(R.id.locationText);
        locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
        locationRTF = new Location("locationManager");
        locationRTF.setLatitude(56.840750);
        locationRTF.setLongitude(60.650750);
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 1);
        }
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
            locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 0, 0, locationListener);
            locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 0, locationListener);
        }
//endregion

        //region classCard
        classImage = findViewById(R.id.classImage);
        classNameText = findViewById(R.id.classNameText);
        classNumberText = findViewById(R.id.classNumberText);
        classTypeText = findViewById(R.id.classTypeText);
        auditoryText = findViewById(R.id.auditoryText);
        lecturerText = findViewById(R.id.lecturerText);
        timeText = findViewById(R.id.timeText);
        classFiling();
        //endregion

        //region Tab
        TabHost tabHost = findViewById(R.id.tabhost);
        tabHost.setup();

        TabHost.TabSpec tabSpec;
        tabSpec = tabHost.newTabSpec("tag1");
        tabSpec.setIndicator("Отметка");
        tabSpec.setContent(R.id.tab1);
        tabHost.addTab(tabSpec);

        tabSpec = tabHost.newTabSpec("tag2");
        tabSpec.setIndicator("Опросы");
        tabSpec.setContent(R.id.tab2);
        tabHost.addTab(tabSpec);

        tabHost.setCurrentTabByTag("tag1");
        //endregion

        //region List
        ArrayList<Map<String, Object>> data = new ArrayList<>(name.length);
        Map<String, Object> m;
        for (int i = 0; i < name.length; i++) {
            m = new HashMap<>();
            m.put(ATTRIBUTE_NAME_NAME, name[i]);
            m.put(ATTRIBUTE_NAME_TYPE, type[i]);
            m.put(ATTRIBUTE_NAME_WHO, who[i]);
            m.put(ATTRIBUTE_NAME_TIME, time[i]);
            m.put(ATTRIBUTE_NAME_IMG, R.mipmap.ic_launcher);
            data.add(m);
        }

        String[] from = {ATTRIBUTE_NAME_NAME, ATTRIBUTE_NAME_TYPE, ATTRIBUTE_NAME_WHO, ATTRIBUTE_NAME_TIME, ATTRIBUTE_NAME_IMG};
        int[] to = {R.id.interviewNameText, R.id.interviewTypeText, R.id.interviewWhoText, R.id.interviewTimeText, R.id.interviewImage};

        SimpleAdapter sAdapter = new SimpleAdapter(this, data, R.layout.interview_card, from, to);

        listViewInterview = findViewById(R.id.listViewInteview);
        listViewInterview.setAdapter(sAdapter);
        //endregion

        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
                    ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 1);
                }

        wifiText = findViewById(R.id.wifiText);
        WifiManager wifiMgr = (WifiManager) getApplicationContext().getSystemService(WIFI_SERVICE);
        wifiInfo = wifiMgr.getConnectionInfo();

        startActivity(new Intent(this, PreviewActivity.class));
    }

    @Override
    protected void onResume() {
        super.onResume();
        SharedPreferences sp = getSharedPreferences("autoriz", MODE_PRIVATE);
        if (sp.contains("task") && sp.getBoolean("task", true)) {
            MyTask mt = new MyTask();
            mt.execute();
        }
        SharedPreferences.Editor ed = sp.edit();
        ed.putBoolean("task", false);
        ed.apply();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.checkButton:
                if (wifiInfo == null){
                    Toast.makeText(this, "Нет подключения", Toast.LENGTH_LONG).show();
                    return;
                }
                if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
                    ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 1);
                    return;
                }
                if (locationStudent != null && locationStudent.distanceTo(locationRTF) > 100)
                    Toast.makeText(this, "СРОЧНО НА ПАРУ", Toast.LENGTH_LONG).show();
                else if (locationStudent != null)
                    Toast.makeText(this, "ЗНАНИЕ - СИЛА", Toast.LENGTH_LONG).show();
                locationText.setText(formatLocation(locationStudent));
                wifiText.setText(wifiInfo.getSSID());
                break;
            case R.id.exitButton:
                SharedPreferences sh = getSharedPreferences("autoriz", MODE_PRIVATE);
                SharedPreferences.Editor e = sh.edit();
                e.putBoolean("auton", false);
                e.apply();
                startActivity(new Intent(this, LoginActivity.class));
                break;
        }
    }

    //region methods
    public void classFiling() {
        classNameText.setText("Математика");
        classNumberText.setText("3");
        classTypeText.setText("Лекция");
        auditoryText.setText("ГУК-404");
        lecturerText.setText("Рыжкова Н. Г.");
        timeText.setText("12:00");
    }

    private LocationListener locationListener = new LocationListener() {
        @Override
        public void onLocationChanged(Location location) {
            locationStudent = location;
        }

        @Override
        public void onStatusChanged(String provider, int status, Bundle extras) {
        }

        @Override
        public void onProviderEnabled(String provider) {
            if (ActivityCompat.checkSelfPermission(getBaseContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
                return;
            }
            locationManager.getLastKnownLocation(provider);
        }

        @Override
        public void onProviderDisabled(String provider) {
        }
    };

    private String formatLocation(Location location) {
        if (location == null)
            return "";
        return String.format("lat = %1$.6f, lon = %2$.6f", location.getLatitude(), location.getLongitude());
    }

    public class MyTask extends AsyncTask<Void, Void, Void> {

        @Override
        protected void onPreExecute(){
            super.onPreExecute();
            checkButton.setBackgroundColor(getResources().getColor(R.color.colorOff));
            checkButton.setEnabled(false);
        }

        @Override
        protected Void doInBackground(Void... params) {
            while(locationStudent == null){
                try {
                    TimeUnit.SECONDS.sleep(1);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
            return null;
        }

        @Override
        protected void onPostExecute(Void result) {
            super.onPostExecute(result);
            checkButton.setBackgroundColor(getResources().getColor(R.color.colorOn));
            checkButton.setEnabled(true);
        }
    }
//endregion
}

File: app/src/main/java/com/example/imhere/MainActivity.kt
package com.example.imhere

import android.Manifest
import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.location.Location
import android.location.LocationListener
import android.location.LocationManager
import android.net.wifi.WifiInfo
import android.net.wifi.WifiManager
import android.os.Bundle
import android.view.View
import android.widget.*
import android.widget.TabHost.TabSpec
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import kotlinx.coroutines.*
import java.util.*


class MainActivity : AppCompatActivity(), View.OnClickListener {
    //region Arrays For List View
    var name = arrayOf("Как дела", "Универ", "Преподы", "Кто ты", "За путина", "Опрос", "Радик", "Радик")
    var type = arrayOf("временный", "постоянный", "временный", "постоянный", "временный", "постоянный", "временный", "временный")
    var who = arrayOf("радик", "Универ", "Преподы", "союз", "деканат", "декан", "преподы", "преподы")
    var time = arrayOf("12:00", "не ограничено", "12:00", "не ограничено", "12:00", "не ограничено", "12:00", "12:00")

    //endregion
    //region Property Declaration
    private var progressBar: ProgressBar? = null
    var listViewInterview: ListView? = null
    private lateinit var checkButton: Button
    private lateinit var exitButton: Button
    var locationText: TextView? = null
    var wifiText: TextView? = null
    var classImage: ImageView? = null
    var classNameText: TextView? = null
    var classNumberText: TextView? = null
    var classTypeText: TextView? = null
    var auditoryText: TextView? = null
    var lecturerText: TextView? = null
    var timeText: TextView? = null
    var locationManager: LocationManager? = null
    var locationStudent: Location? = null
    var locationRTF: Location? = null
    var requestLocationUpdateMade = false
    private var locationListener: LocationListener = object : LocationListener {
        override fun onLocationChanged(location: Location) {
            locationStudent = location
        }

        override fun onStatusChanged(provider: String, status: Int, extras: Bundle) {}
        override fun onProviderEnabled(provider: String) {
            if (ActivityCompat.checkSelfPermission(baseContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) return
            locationManager!!.getLastKnownLocation(provider)
        }

        override fun onProviderDisabled(provider: String) {}
    }
    var wifiMgr: WifiManager? = null

    //endregion
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //region Property Initializing
        progressBar = findViewById(R.id.progressBar)
        checkButton = findViewById(R.id.checkButton) //Инициализация основных View.
        exitButton = findViewById(R.id.exitButton)
        listViewInterview = findViewById(R.id.listViewInteview)
        wifiText = findViewById(R.id.wifiText)
        locationText = findViewById(R.id.locationText)
        classImage = findViewById(R.id.classImage) //Инициализация элементов карточки в информацией
        classNameText = findViewById(R.id.classNameText) //о ближайшей паре. Для нее нужно спрасить
        classNumberText = findViewById(R.id.classNumberText) //расписание из ЛК.
        classTypeText = findViewById(R.id.classTypeText)
        auditoryText = findViewById(R.id.auditoryText)
        lecturerText = findViewById(R.id.lecturerText)
        timeText = findViewById(R.id.timeText)
        //endregion
        checkButton.setOnClickListener(this)
        exitButton.setOnClickListener(this)
        progressBar!!.visibility = View.INVISIBLE

        //region Location Block
        locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
        locationRTF = Location("locationManager")
        with(locationRTF!!) {
            latitude = 56.840750
            longitude = 60.650750
        }
        ActivityCompat.requestPermissions(this@MainActivity, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
            with(locationManager!!) {
                requestLocationUpdates(LocationManager.GPS_PROVIDER, 0, 0f, locationListener)
                requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 0f, locationListener)
            }
            requestLocationUpdateMade = true
        } else ActivityCompat.requestPermissions(this@MainActivity, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
        //endregion
        wifiMgr = applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager?
        classCardCreate() //Заполнение всех View
        tabHostCreate()
        listViewCreate()
        startActivity(Intent(this, PreviewActivity::class.java)) //Запуск превью
    }

    @SuppressLint("ResourceAsColor")
    override fun onClick(v: View) {
        when (v.id) {
            R.id.checkButton -> {
                GlobalScope.launch(Dispatchers.Main) {
                    if (ActivityCompat.checkSelfPermission(this@MainActivity, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
                        ActivityCompat.requestPermissions(this@MainActivity, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
                        return@launch
                    }
                    if (!requestLocationUpdateMade) makeRequestLocationUpdate()
                    progressBar!!.visibility = View.VISIBLE
                    checkButton.isClickable = false
                    withContext(Dispatchers.IO){ while(locationStudent == null) delay(50) }
                    if (ActivityCompat.checkSelfPermission(this@MainActivity, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
                             ActivityCompat.requestPermissions(this@MainActivity, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), 1)
                             return@launch
                         }
                    if (locationStudent!!.distanceTo(locationRTF) > 100)
                        Toast.makeText(this@MainActivity, "СРОЧНО НА ПАРУ", Toast.LENGTH_LONG).show()
                    else Toast.makeText(this@MainActivity, "ЗНАНИЕ - СИЛА", Toast.LENGTH_LONG).show()
                    locationText!!.text = formatLocation(locationStudent)
                    wifiText!!.text = wifiMgr?.connectionInfo?.ssid
                    progressBar!!.visibility = View.INVISIBLE
                    checkButton.isClickable = true
                    }
                 }
            R.id.exitButton -> {
                startActivity(Intent(this, AddInterviewActivity::class.java))
                /*val sh = getSharedPreferences("authentication", Context.MODE_PRIVATE)
                val e = sh.edit()
                e.putBoolean("authentication", false)
                e.apply()
                startActivity(Intent(this, LoginActivity::class.java))*/
            }
        }
    }

    //region Methods For Filling Layout
    private fun tabHostCreate() {
        val tabHost = findViewById<TabHost>(R.id.tabhost)
        tabHost.setup()
        var tabSpec: TabSpec
        tabSpec = tabHost.newTabSpec("tag1")
        tabSpec.setIndicator("Отметка")
        tabSpec.setContent(R.id.tab1)
        tabHost.addTab(tabSpec)
        tabSpec = tabHost.newTabSpec("tag2")
        tabSpec.setIndicator("Опросы")
        tabSpec.setContent(R.id.tab2)
        tabHost.addTab(tabSpec)
        tabHost.setCurrentTabByTag("tag1")
    }

    private fun listViewCreate() {
        val data = ArrayList<Map<String, Any?>>(name.size)
        var m: MutableMap<String, Any?>
        val from = arrayOf(getString(R.string.attribute_name_name),
                getString(R.string.attribute_name_type),
                getString(R.string.attribute_name_who),
                getString(R.string.attribute_name_time),
                getString(R.string.attribute_name_img))
        val to = intArrayOf(R.id.interviewNameText,
                R.id.interviewTypeText,
                R.id.interviewWhoText,
                R.id.interviewTimeText,
                R.id.interviewImage)
        for (i in name.indices) {
            m = HashMap()
            m[from[0]] = name[i]
            m[from[1]] = type[i]
            m[from[2]] = who[i]
            m[from[3]] = time[i]
            m[from[4]] = R.mipmap.ic_launcher
            data.add(m)
        }
        listViewInterview!!.adapter = SimpleAdapter(this, data, R.layout.interview_card, from, to)
    }

    private fun classCardCreate() {
        classNameText!!.text = "Математика"
        classNumberText!!.text = "3"
        classTypeText!!.text = "Лекция"
        auditoryText!!.text = "ГУК-404"
        lecturerText!!.text = "Рыжкова Н. Г."
        timeText!!.text = "12:00"
    }

    //endregion
    //region Auxiliary Methods
    private fun formatLocation(location: Location?): String {
        return if (location == null) "" else String.format("lat = %1$.6f, lon = %2$.6f", location.latitude, location.longitude)
    }

    @SuppressLint("MissingPermission")
    private fun makeRequestLocationUpdate(){
        with(locationManager!!) {
            requestLocationUpdates(LocationManager.GPS_PROVIDER, 0, 0f, locationListener)
            requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 0f, locationListener)
        }
        requestLocationUpdateMade = true
    }
    //endregion
}
